src/vmspawn/vmspawn.c

```diff
@@ -538,6 +624,19 @@ static int run_virtual_machine(void) {
         if (!cmdline)
                 return log_oom();
 
+        /* if we are going to be starting any units with state then create our runtime dir */
+        if (arg_tpm != 0) {
+                r = runtime_directory(&arg_runtime_directory, arg_privileged ? RUNTIME_SCOPE_SYSTEM : RUNTIME_SCOPE_USER, "systemd/vmspawn");
+                if (r < 0)
+                        return log_error_errno(r, "Failed to lookup runtime directory: %m");
+                if (r) {
```

Comment:
Can you add a brief comment here that explains what this check does, i.e. `/* r > 0 means we need to create our own runtime dir */`